### ISSUE TITLE
Tickets are having duplicate attachments in versions 1.8 and above when sending emails using Gmail

### ISSUE THREAD

**Issue by kmoksha** [osTicket/osTicket#1743]
I have installed new Osticket 1.9.5.1 on my site and have linked a support email with Osticket via piping for tickets to be generated.

But when user emails with attachments are received from Gmail, duplicate attachments are seen in the generated ticket. ( Noticed that 2 keys are generated for the same attachment ; see screenshots at the end) Why is it ? 

Where are the attached files sent by the users stored by default ?

I am linking the webmail with Osticket via piping., Mail Sending in settings is showing via PHP mail function, Php version is 5.4.35 , Mysql version is 5.5.40 , Operating system is Windows 7

Also, on clicking on the ticket attachment, the attachment does not download like it used to in versions 1.7 and earlier but they open up in the browser. That will mean extra work for the staff.

![duplicate attachment with versions 1 8 osticket with gmail 1](https://cloud.githubusercontent.com/assets/8469955/6242629/ffd504d0-b758-11e4-83b4-1d441ba84263.jpg)

![duplicate attachment with versions 1 8 osticket with gmail 2](https://cloud.githubusercontent.com/assets/8469955/6242630/ffdbfaec-b758-11e4-8cc2-90b6f221de75.jpg)

**Comment by greezybacon** [comment]
@kmoksha Notice the keys are different. Do you have a raw version of the email you could post so I could see what's going on? I agree the inline attachment download is odd for the attachment links under the thread entry.

**Comment by kmoksha** [comment]
@ greezybacon

sorry, No longer have the email copy of the mail of which screenshot is posted.

Yes, keys are different but the attachments are the same.

Did you try to duplicate the error ? Can you post screenshots of the same ?

Can anyone else try to look into this also and tell whether they are facing same problem with gmail generated tickets ?

**Comment by greezybacon** [comment]
@kmoksha we primarily use gmail and haven't seen this. Also since v1.9.1, the system makes it best efforts to deduplicate file contents. In fact there are other issues open because we reduplicate too aggressively. My guess is that the email really had two attachments with the same name and (slightly) differing contents. 

In fact the key is actually generated based on the SHA1 hash of the contents. Then it is prefixed with a random code. Therefore, the ending of the keys would also be the same if the contents had the same SHA1 hash.

**Comment by kmoksha** [comment]
@greezybacon

I have shown in the screenshot the mail attachments is the same which got duplicated. I have replicated the issue several times and have now done it again. 

The mail has only one attachment but the generated ticket has TWO attachments which are duplicate. If you want, I can forward you the raw mail for that if you give me your mail id.

Attaching the error which I replicated just now.
![osticket duplicate ticket error 19 march 2015](https://cloud.githubusercontent.com/assets/8469955/6733522/ac566e34-ce78-11e4-823e-c35cb0d619aa.png)
